File: ios/ApplifierImpact/ApplifierImpactZone/ApplifierImpactIncentivizedZone.m
//
//  ApplifierImpactIncentivizedZone.m
//  ApplifierImpact
//
//  Created by Ville Orkas on 10/1/13.
//  Copyright (c) 2013 Applifier. All rights reserved.
//

#import "ApplifierImpactIncentivizedZone.h"
#import "ApplifierImpactConstants.h"
#import "ApplifierImpactRewardItem.h"

@interface ApplifierImpactIncentivizedZone ()

@property (nonatomic, strong) ApplifierImpactRewardItemManager *_itemManager;

@end

@implementation ApplifierImpactIncentivizedZone

- (id)initWithData:(NSDictionary *)options {
  self = [super initWithData:options];
  if(self) {
    id items = [options objectForKey:kApplifierImpactZoneRewardItemsKey];
    NSMutableDictionary * itemsDictionary = [[NSMutableDictionary alloc] init];
    [items enumerateObjectsUsingBlock:^(id rawItem, NSUInteger idx, BOOL *stop) {
      id item = [[ApplifierImpactRewardItem alloc] initWithData:rawItem];
      [itemsDictionary setObject:item forKey:[item key]];
    }];
    id defaultItem = [[ApplifierImpactRewardItem alloc] initWithData:[options objectForKey:kApplifierImpactZoneDefaultRewardItemKey]];
    self._itemManager = [[ApplifierImpactRewardItemManager alloc] initWithItems:itemsDictionary defaultItem:defaultItem];
  }
  return self;
}

- (BOOL)isIncentivized {
  return TRUE;
}

- (ApplifierImpactRewardItemManager *)itemManager {
  return self._itemManager;
}

@end

File: android/sources/src/com/applifier/impact/android/zone/ApplifierImpactIncentivizedZone.java
package com.applifier.impact.android.zone;

import org.json.JSONException;
import org.json.JSONObject;

import com.applifier.impact.android.item.ApplifierImpactRewardItem;
import com.applifier.impact.android.item.ApplifierImpactRewardItemManager;
import com.applifier.impact.android.properties.ApplifierImpactConstants;

public class ApplifierImpactIncentivizedZone extends ApplifierImpactZone {

	private ApplifierImpactRewardItemManager _rewardItems = null;
	
	public ApplifierImpactIncentivizedZone(JSONObject zoneObject) throws JSONException {
		super(zoneObject);
		ApplifierImpactRewardItem defaultItem = new ApplifierImpactRewardItem(zoneObject.getJSONObject(ApplifierImpactConstants.IMPACT_ZONE_DEFAULT_REWARD_ITEM_KEY));
		_rewardItems = new ApplifierImpactRewardItemManager(zoneObject.getJSONArray(ApplifierImpactConstants.IMPACT_ZONE_REWARD_ITEMS_KEY), defaultItem.getKey());
	}
	
	@Override
	public boolean isIncentivized() {
		return true;
	}
	
	public ApplifierImpactRewardItemManager itemManager() {
		return _rewardItems;
	}

}
